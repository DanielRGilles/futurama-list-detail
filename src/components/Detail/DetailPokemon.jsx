import React from 'react'
import styles from '../../views/Detail/Detail.css'
import colors from '../List/CharacterList.css'
export default function DetailPokemon({pokemon}) {
    const {image, type, name, color, hp, height, weight, ability1, ability2} = pokemon;
    return (
        <>
             <section className={styles.container + ' ' + colors[`${type}`]}>
                <img src={image}/>
                <h3>{name}</h3>
                <p>HP: {hp}</p>
                <p>Height: {height} </p>
                <p>Weight: {weight} pounds</p>
                <p>Ability 1:{ability1}</p>
                <p>Ability 2:{ability2}</p>
                <p>Color: {color}</p>

            </section>
            {console.log(pokemon)}
        </>
    )
}
