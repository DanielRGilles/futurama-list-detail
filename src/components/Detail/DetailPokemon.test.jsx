import {render } from  '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DetailPokemon from './DetailPokemon';


describe('Header component test', () => {
    const pokemon = {ability1: "snow-warning",
    ability2: "NA",
    color: "#78C850",
    height: 22,
    hp: 90,
    id: "5ff4fb7cd89993a89cc6563f",
    image: "http://assets.pokemon.com/assets/cms2/img/pokedex/full/460.png",
    name: "abomasnow",
    type: "grass",
    weight: 1355}
    
    it('renders a detail view of  a pokemon', () => {
       const { container } = render(
            <MemoryRouter>
                
                     <DetailPokemon pokemon={pokemon}/>
               
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
         
    })

})