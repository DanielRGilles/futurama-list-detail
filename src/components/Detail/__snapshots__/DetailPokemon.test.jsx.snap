// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test renders a detail view of  a pokemon 1`] = `
<div>
  <section
    class="container grass"
  >
    <img
      src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/460.png"
    />
    <h3>
      abomasnow
    </h3>
    <p>
      HP: 
      90
    </p>
    <p>
      Height: 
      22
       
    </p>
    <p>
      Weight: 
      1355
       pounds
    </p>
    <p>
      Ability 1:
      snow-warning
    </p>
    <p>
      Ability 2:
      NA
    </p>
    <p>
      Color: 
      #78C850
    </p>
  </section>
</div>
`;
